sphinxcontrib/blockdiag.py

```diff
@@ -312,9 +333,13 @@ def setup(app):
     app.add_config_value('blockdiag_antialias', False, 'html')
     app.add_config_value('blockdiag_transparency', True, 'html')
     app.add_config_value('blockdiag_debug', False, 'html')
-    app.add_config_value('blockdiag_html_image_format', 'PNG', 'html')
-    app.add_config_value('blockdiag_tex_image_format', None, 'html')  # backward compatibility for 1.3.1
-    app.add_config_value('blockdiag_latex_image_format', 'PNG', 'html')
+    app.add_config_value('blockdiag_image_format', None, 'env')
```

Comment:
IMO, it would be better to use a function as a generator.
```
def default_blockdiag_image_format(config):
    if config.blockdiag_latex_image_format
        latex_image_format = config.blockdiag_latex_image_format
    else:
        latex_image_format = config.blockdiag_tex_image_format

    return {
        'html': config.blockdiag_html_image_format,
        'slides': config.blockdiag_html_image_format,
        'latex': latex_image_format,
    }

def setup(app):
    app.add_config_value('blockdiag_image_format', default_blockdiag_image_format, 'env')
```

Then, you don't need to care about the old configurations because they will be converted to the new style automatically.
If users configure the image formats via the new style one, `blockdiag_image_format` will be overridden.

Reply:
Updated to use the recommended generator and remove the format fallbacks inside `get_image_format_for`.